core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -1338,15 +1342,35 @@ private void deleteSubjectCompatibility(String subject) throws StoreException {
 
   public Mode getMode(String subject) throws SchemaRegistryStoreException {
     try {
-      return lookupCache.mode(subject, false, defaultMode);
+      SchemaRegistryValue globalModeValue = this.kafkaStore.get(new ModeKey(null));
+      Mode globalMode = globalModeValue != null ? ((ModeValue) globalModeValue).getMode() : null;
+      Mode subjectMode = lookupCache.mode(subject, false, defaultMode);
+
+      if (globalMode == Mode.READONLY_OVERRIDE) {
+        return globalMode;
+      } else if (subjectMode != null) {
+        return subjectMode;
+      } else {
+        return globalMode != null ? globalMode : defaultMode;
+      }
     } catch (StoreException e) {
       throw new SchemaRegistryStoreException("Failed to write new config value to the store", e);
     }
   }
 
   private Mode getModeInScope(String subject) throws SchemaRegistryStoreException {
     try {
-      return lookupCache.mode(subject, true, defaultMode);
+      SchemaRegistryValue globalModeValue = this.kafkaStore.get(new ModeKey(null));
+      Mode globalMode = globalModeValue != null ? ((ModeValue) globalModeValue).getMode() : null;
```

Comment:
I think this can be replaced with `Mode globalMode = lookupCache.mode(null, false, defaultMode)`